src/Core/Models/Api/Request/Accounts/VerifyPasswordRequestModel.cs

```diff
@@ -1,12 +1,27 @@
-﻿using System;
+﻿using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
 
 namespace Bit.Core.Models.Api
 {
-    public class VerifyPasswordRequestModel
+    public class VerifyPasswordRequestModel : IValidatableObject
     {
-        [Required]
         [StringLength(300)]
         public string MasterPasswordHash { get; set; }
+        public string OTP { get; set; }
+
+        public string Secret => SuppliedMasterPassword() ? MasterPasswordHash : OTP;
+
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
+            if (string.IsNullOrEmpty(MasterPasswordHash) && string.IsNullOrEmpty(OTP))
+            {
+                yield return new ValidationResult("MasterPasswordHash or OTP must be supplied.");
```

Comment:
Validators in asp.net typically follows this flow. It's a trick to avoid having to compute the whole `IEnumerable` when an item often only wants to know the first. https://docs.microsoft.com/en-us/aspnet/core/mvc/models/validation?view=aspnetcore-6.0#ivalidatableobject